File: week-04/java/day-1/src/PrintEachLine.java
import java.io.IOException;
import java.nio.file.Path;
import java.nio.file.Paths;
import java.nio.file.Files;
import java.util.ArrayList;
import java.util.List;

public class PrintEachLine {
    public static void main(String[] args) {
        // Write a program that opens a file called "my-file.txt", then prints
        // each of the lines form the file.
        // If the program is unable to read the file (for example it does not exists),
        // then it should print an error message like: "Unable to read file: my-file.txt"

        List<String> fileContent = readFile("new-file.txt");

        for(int i = 0; i < fileContent.size(); i++){
            System.out.println(fileContent.get(i));
        }
    }

    public static List<String> readFile(String path){
        try{
            Path filePath = Paths.get(path);
            List<String> lines = Files.readAllLines(filePath);
            return lines;
        } catch(IOException e) {
            System.out.println("Uh-oh, could not read the file!");
            return null;
        }
    }
}

File: week-04/typescript/day-1/print-each-line.ts
// Write a program that opens a file called 'my-file.txt', then prints
// each of lines form the file.
// If the program is unable to read the file (for example it does not exists),
// then it should print an error message like: 'Unable to read file: my-file.txt'

const fs = require('fs');

try {
  let fileContent = fs.readFileSync('files/my-file.txt', 'utf-8');
  console.log(fileContent);
} catch (err) {
  console.log('Unable to read file: my-file.txt');
}